packages/vitest/src/node/reporters/json.ts

```diff
@@ -124,12 +124,7 @@ export class JsonReporter implements Reporter {
         assertionResults,
         startTime,
         endTime,
-        status: tests.every(t =>
-          t.result?.state === 'pass'
-           || t.result?.state === 'skip'
-            || t.result?.state === 'todo')
-          ? 'passed'
-          : 'failed',
+        status: StatusMap[file.result?.state || file.mode] || 'skipped',
```

Comment:
Unfortunately it isn't. In `jest` test file status can only be `passed` or `failed`. And it treats any test file with non-failing tests as `passed`. Check: [packages/jest-test-result/src/types.ts](https://github.com/facebook/jest/blob/main/packages/jest-test-result/src/types.ts) and [packages/jest-test-result/src/formatTestResults.ts](https://github.com/facebook/jest/blob/main/packages/jest-test-result/src/formatTestResults.ts)

So I see the following options:

1. Introduce a breaking change, by changing `status` type and the way it works. In my opinion having `skipped` status for test file is much more meaningful than treating it as `passed`. Moreover, it is consistent with default and verbose `vitest` reporters.

3. We mimic jest reporter and treat the test file with all tests skipped as `passed`, which is in my opinion misleading. **Also**, if we opt for the full consistency with `jest` we would need to change more things. For example, `jest` counts the test file with all skipped tests as `skipped` (even though the `status` field of this test file is `passed`). `vitest` on the other hand treats it as passed, which results in different values for `numPendingTests` and `numPassedTests` for both reporters.

For instance, this test:
```
it.skip('2 + 3 = 5', () => {
  expect(2 + 3).toBe(5)
})

it.skip('3 + 3 = 6', () => {
  expect(3 + 3).toBe(6)
})
```
has the following output in `jest`:
```json
{
  "numFailedTestSuites": 0,
  "numFailedTests": 0,
  "numPassedTestSuites": 0,
  "numPassedTests": 0,
  "numPendingTestSuites": 1,
  "numPendingTests": 2,
  "numRuntimeErrorTestSuites": 0,
  "numTodoTests": 0,
  "numTotalTestSuites": 1,
  "numTotalTests": 2,
  "openHandles": [],
  "snapshot": {
    "added": 0,
    "didUpdate": false,
    "failure": false,
    "filesAdded": 0,
    "filesRemoved": 0,
    "filesRemovedList": [],
    "filesUnmatched": 0,
    "filesUpdated": 0,
    "matched": 0,
    "total": 0,
    "unchecked": 0,
    "uncheckedKeysByFile": [],
    "unmatched": 0,
    "updated": 0
  },
  "startTime": 1670076050633,
  "success": true,
  "testResults": [
    {
      "assertionResults": [
        {
          "ancestorTitles": [],
          "duration": null,
          "failureDetails": [],
          "failureMessages": [],
          "fullName": "2 + 3 = 5",
          "invocations": 1,
          "location": null,
          "numPassingAsserts": 0,
          "retryReasons": [],
          "status": "pending",
          "title": "2 + 3 = 5"
        },
        {
          "ancestorTitles": [],
          "duration": null,
          "failureDetails": [],
          "failureMessages": [],
          "fullName": "3 + 3 = 6",
          "invocations": 1,
          "location": null,
          "numPassingAsserts": 0,
          "retryReasons": [],
          "status": "pending",
          "title": "3 + 3 = 6"
        }
      ],
      "endTime": 1670076051183,
      "message": "",
      "name": "C:\vue-project\tests\test.spec.ts",
      "startTime": 1670076050755,
      "status": "passed",
      "summary": ""
    }
  ],
  "wasInterrupted": false
}
```
and the following in `vitest`:
```json
{
  "numFailedTestSuites": 0,
  "numFailedTests": 0,
  "numPassedTestSuites": 1,
  "numPassedTests": 2,
  "numPendingTestSuites": 0,
  "numPendingTests": 0,
  "numTodoTests": 0,
  "numTotalTestSuites": 1,
  "numTotalTests": 2,
  "startTime": 1670076324873,
  "success": true,
  "testResults": [
    {
      "assertionResults": [
        {
          "ancestorTitles": [
            ""
          ],
          "fullName": " 2 + 3 = 5",
          "status": "skipped",
          "title": "2 + 3 = 5",
          "failureMessages": []
        },
        {
          "ancestorTitles": [
            ""
          ],
          "fullName": " 3 + 3 = 6",
          "status": "skipped",
          "title": "3 + 3 = 6",
          "failureMessages": []
        }
      ],
      "startTime": 1670076324873,
      "endTime": 1670076324873,
      "status": "failed",
      "message": "",
      "name": "C:/vue-project/tests/test.spec.ts"
    }
  ]
}
```

Which option should we choose?
In case of option 1 - how should I proceed? Is marking PR as breaking enough, or should we update docs, etc.?
In case of option 2 - should this PR focus only on fixing test file `status`, or should I fix other inconsistencies with `jest` json reporter?

Reply:
The idea of json reporter was to be compatible with jest one because of tooling.

This PR can be merged, if possible status values stay as they were before. And only the bug is fixed. If you want to fix more inconsistencies, you are welcome to, but it's not required.

Reply:
Okay, I see. So if I wanted to have the behavior I described I guess would need to implement custom reporter for my project.

To keep things simple I fixed the `status` property for now. If I have more time to analyze how `jest` calculates all other fields in the output, I will fix it in separate PR.

However I think it makes sense to leave here the test case with "skipped only tests" to emphasize that this behavior is intentional.